integration-cli/docker_cli_build_test.go

```diff
@@ -6740,3 +6740,37 @@ func (s *DockerSuite) TestBuildLabelsOverride(c *check.C) {
 	}
 
 }
+
+// #20083
+func (s *DockerSuite) TestBuildDockerignoreComment(c *check.C) {
+	name := "testbuilddockerignorecleanpaths"
+	dockerfile := `
+        FROM busybox
+        ADD . /tmp/
+        RUN sh -c "(! ls /tmp/foo) && (! ls /tmp/foo2) && (ls /tmp/dir1/foo) && (ls /tmp/#1) && (! ls /tmp/#2)"`
+	ctx, err := fakeContext(dockerfile, map[string]string{
+		"foo":      "foo",
+		"foo2":     "foo2",
+		"dir1/foo": "foo in dir1",
+		"#1":       "# file 1",
+		"#2":       "# file 2",
+		".dockerignore": `# Visual C++ cache files
+# because we have git ;-)
+# The above comment is from #20083
+./foo
+#dir1/foo
+./foo2
+# The following is considered as comment as # is at the beginning
+#1
+# The following is not considered as comment as # is not at the beginning
+  #2
```

Comment:
Yes, it is a bit odd, but we have allowed leading spaces before filenames already, so we don't really want to change this behaviour, and while it does not provide a way to ignore files with spaces at the start, it does allow a workaround to ignore files with # at the start...!